presto-hudi/src/main/java/com/facebook/presto/hudi/split/HudiPartitionSplitGenerator.java

```diff
@@ -0,0 +1,156 @@
+            HudiTableLayoutHandle layout,
+            HoodieTableFileSystemView fsView,
+            AsyncQueue<ConnectorSplit> asyncQueue,
+            Queue<String> concurrentPartitionQueue,
+            String latestInstant)
+    {
+        this.metastore = requireNonNull(metastore, "metastore is null");
+        this.metastoreContext = toMetastoreContext(requireNonNull(session, "session is null"));
+        this.layout = requireNonNull(layout, "layout is null");
+        this.table = layout.getTable();
+        this.tablePath = new Path(table.getPath());
+        this.fsView = requireNonNull(fsView, "fsView is null");
+        this.asyncQueue = requireNonNull(asyncQueue, "asyncQueue is null");
+        this.concurrentPartitionQueue = requireNonNull(concurrentPartitionQueue, "concurrentPartitionQueue is null");
+        this.latestInstant = requireNonNull(latestInstant, "latestInstant is null");
+        this.splitWeightProvider = createSplitWeightProvider(requireNonNull(session, "session is null"));
+    }
+
+    @Override
+    public void run()
+    {
+        HoodieTimer timer = new HoodieTimer().startTimer();
+
+        while (!concurrentPartitionQueue.isEmpty()) {
+            String partitionName = concurrentPartitionQueue.poll();
+
+            if (partitionName != null) {
+                generateSplitsFromPartition(partitionName);
+            }
+        }
+        log.debug("HudiPartitionSplitGenerator %s finishes in %d ms", this, timer.endTimer());
+    }
+
+    private void generateSplitsFromPartition(String partitionName)
+    {
+        HudiPartition hudiPartition = getHudiPartition(metastore, metastoreContext, layout, partitionName);
+        Path partitionPath = new Path(hudiPartition.getStorage().getLocation());
+        String relativePartitionPath = FSUtils.getRelativePartitionPath(tablePath, partitionPath);
+        fsView.getLatestFileSlicesBeforeOrOn(relativePartitionPath, latestInstant, false)
+                .map(fileSlice -> createHudiSplit(table, fileSlice, latestInstant, hudiPartition, splitWeightProvider))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .forEach(asyncQueue::offer);
+    }
+
+    private Optional<HudiSplit> createHudiSplit(
+            HudiTableHandle table,
+            FileSlice slice,
+            String timestamp,
+            HudiPartition partition,
+            HudiSplitWeightProvider splitWeightProvider)
+    {
+        HudiFile hudiFile = slice.getBaseFile().map(f -> new HudiFile(f.getPath(), 0, f.getFileLen())).orElse(null);
+        if (null == hudiFile && table.getTableType() == HudiTableType.COW) {
+            return Optional.empty();
+        }
+        List<HudiFile> logFiles = slice.getLogFiles()
+                .map(logFile -> new HudiFile(logFile.getPath().toString(), 0, logFile.getFileSize()))
+                .collect(toImmutableList());
+        long sizeInBytes = hudiFile != null ? hudiFile.getLength()
```

Comment:
should n't we sum base + log instead of picking base or sum of log files?

Reply:
it's a good point. i'll correct it.